### PR TITLE
Add prettier

### PR DESCRIPTION
**Pull request by austingardner**
## Description
This PR adds prettier to the skin repo. 
I added a package called lint-staged which also required another package called husky to dev dependencies. These packages run a pre-commit hook that runs prettier as well as our linter. 

I also added eslint-config-prettier, stylelint-prettier and stylelint-config-prettier to dev dependencies. These libraries adjust eslint and stylelint appropriately to turn off style settings that could interfere with prettier. This prevents a situation where prettier could re-format a file and then that file is in violation of an eslint rule. 

Importantly, these libraries will still keep lint rules that are important to preventing bugs.

We will probably need to slightly adjust the configurations of these things over time to suit our tastes. But for now this adds a comprehensive solution which provides significant savings in dev time. 

### Instructions
Just install the appropriate prettier plugin for your code editor, such as [Prettier for Atom](https://atom.io/packages/prettier-atom), and enable "Format on Save" to see prettier in action. 

Alternatively, just wait until you make a commit and lint-staged will run prettier only on the files that you have changed. 

## Context
This will help us ensure uniform styles and fewer broken CI builds because of styling issues.

### CODE REVIEW

**Comment by agliga** [comment]
I think this looks good. 
My only issue is to try to make it follow the current styles and standards we have
Ideally we should have a very little amount of file changes in this imo. 

Should we also do this precommit? Prepush wont change the files.

**Comment by austingardner** [comment]
So adding husky and lint-staged means that prettier will run on all of the staged files. 

So basically prettier will run whenever a new commit is made. I couldn't get the 'pre-push' or 'pre-commit' hooks to work as expected, which is why I added husky and lint-staged.

**Comment by austingardner** [comment]
I will look into our styles and see if I can configure prettier to make fewer changes.

However, many of the changes are from our own inconsistencies, i.e. using both single and double quotes in the same file.

File: stepper.browser.json [hunk 1]
```diff
@@ -1,5 +1,3 @@
 {
-    "dependencies": [
-         "require: ./stepper"
-    ]
+    "dependencies": ["require: ./stepper"]
```
Comment (ianmcburnie) [inline stepper.browser.json:2]: Hmm, don't all these json changes go against our eslint rules? I'm pretty sure our coding standard is to not have array blocks declared on the same line like this.

File: package.json [hunk 2]
```diff
@@ -1,115 +1,137 @@
...
+        "eBay MIND Patterns"
+    ],
+    "publishConfig": {
+        "registry": "https://registry.npmjs.org"
+    },
+    "browserslist": [
+        "extends @ebay/browserslist-config"
+    ],
+    "dependencies": {
+        "eslint-plugin-prettier": "^3.3.1"
```
Comment (ianmcburnie) [inline package.json:124]: Should this be a devDependency rather than a dependency?

**Comment by ianmcburnie** [comment]
Not sure if things are working correctly, or perhaps need more configuration. I made a deliberate LESS file change that would break stylelint, then tried to commit. I got an error, but it doesn't tell me the reason or source of the error, only that it prevented an empty-git commit.

Screen Shot 2021-01-22 at 2 29 53 PM

File: package.json (hunk 2 above)
Comment (austingardner) [reply package.json:124]: yes, good catch

File: stepper.browser.json (hunk 1 above)
Comment (austingardner) [reply stepper.browser.json:2]: So the advantage of prettier is that it takes care of styling like this for us, so we don't have to worry about it. 
The `eslint-config-prettier` purposely turns off styling rules like that, so we don't have to discuss code styling during code reviews, etc.

Of course the disadvantage is that sometimes the styling may not be what we're always used to. Let me look and see if this is a configurable option for prettier to have array blocks declared on separate lines as we have before this proposed change.

**Comment by austingardner** [comment]
Hmm I'm curious about the stylelint output not being correct. I made a change that would break stylelint and it failed as expected, then output the stylelint fail text correctly 

Screen Shot 2021-01-27 at 11 39 42 AM

**Comment by austingardner** [comment]
Maybe you just forgot to run `git add` before trying to run the commit? I do that all the time 😂

**Comment by ianmcburnie** [comment]
@austingardner Can you change the target branch to 7.1.0? That way it will be in sync with latest changes and we'll run into less merge conflicts.

**Review by ianmcburnie — APPROVED** [review]
Nice work @austingardner ! This is a great addition to the project! :-)

**Review by agliga — APPROVED** [review]

**Pull request merged**